acstools/utils_findsat_mrt.py

```diff
@@ -1199,10 +1200,54 @@ def update_dq(filename, ext, mask, dqval=16384, verbose=True):
         tailored for ACS/WFC.
     verbose : bool, optional
         Print extra information to the terminal.
+    expand_mask: int, optional
+        Allows the mask to be expanded to match the size of the original mask.
+        This is relevant for masks that were generated used binned versions of
+        the original image
 
     """
     with fits.open(filename, mode='update') as pf:
         dqarr = pf[ext].data
+
+        # check that the old mask and new mask are the same size
+        same_size = mask.shape == dqarr.shape
+
+        if not same_size:
+            LOG.info('Inconsistent mask sizes: \n'
+                     'Input mask: {} \n'
+                     'DQ array: {}'.format(mask.shape, dqarr.shape))
+            if not expand_mask:
+                LOG.warning('Cannot proceed due to size mismatch.\n'
+                            'Set expand_mask=True if the mask was generated '
+                            'with binned data and needs to be enlarged to the '
+                            'original size')
+                return
+            else:
+                LOG.info('Enlarging mask to original size')
+                factor_x = np.floor(dqarr.shape[1]/mask.shape[1]).astype(int)
+                factor_y = np.floor(dqarr.shape[0]/mask.shape[0]).astype(int)
+                mask = block_replicate(mask, [factor_y, factor_x],
+                                       conserve_sum=False)
+
+                # for original data that had odd dimensions before binning, the
+                # mask still will not match the original. Duplicate end
+                # rows/columns in this case
+
+                same_size = mask.shape == dqarr.shape
```

Comment:
Instead of doing it twice, is there a smarter way to check for odd/even at the start and then go different logic routes from there?

Reply:
I cleaned this up a bit, but it's more like an extra step if the images don't divide evenly, rather than completely different logic